# HG changeset patch
# User Rik <rik@octave.org>
# Date 1414523212 25200
#      Tue Oct 28 12:06:52 2014 -0700
# Branch gui-release
# Node ID cd8cf41f130ddb3d2241054c2e637b632c8a21d6
# Parent  0e077a29854f2986dfbdcf39aba563e3970069c7
# Parent  d60b44acfff765ebd061b0ac1ab97ec482548c5b
maint: Periodic merge of stable to gui-release.

diff --git a/liboctave/array/Range.h b/liboctave/array/Range.h
--- a/liboctave/array/Range.h
+++ b/liboctave/array/Range.h
@@ -49,17 +49,17 @@ public:
     : rng_base (b), rng_limit (l), rng_inc (i),
       rng_nelem (nelem_internal ()), cache () { }
 
   // For operators' usage (to preserve element count).
   Range (double b, double i, octave_idx_type n)
     : rng_base (b), rng_limit (b + (n-1) * i), rng_inc (i),
       rng_nelem (n), cache ()
   {
-    if (! xfinite (b) || ! xfinite (i) | ! xfinite (rng_limit))
+    if (! xfinite (b) || ! xfinite (i) || ! xfinite (rng_limit))
       rng_nelem = -2;
   }
 
   double base (void) const { return rng_base; }
   double limit (void) const { return rng_limit; }
   double inc (void) const { return rng_inc; }
   octave_idx_type nelem (void) const { return rng_nelem; }
 
